### PR TITLE
Integrate Google Tag Manager

### PR DESCRIPTION
**Pull request by jazzsequence**

### CODE REVIEW

**Comment by jazzsequence** [comment]
I'm not sure if I haven't done this right or if I just don't know how to configure the triggers in Google Tag Manager correctly, but I can't get this working and I'm not really sure what's wrong. :/

**Comment by roborourke** [comment]
I looked a bit further into the Didomi method and it's pretty good and might simplify things for us a bit:

https://developers.didomi.io/cmp/web-sdk/third-parties/custom-integrations/tag-managers/google-tag-manager

Essentially on a `consent-change` event we can set a variable that is a serialised string of the consented categories, the triggers can then check if the variable contains the target category string.

**Review by roborourke — COMMENTED** [review]
It's along the right lines, I've added inline comments to explain some of the tag manager stuff. Main thing was that you needed to push an object with an event property into the data layer variable.

I was also thinking further about this and we could avoid the `wp_localize_script()` and making an extra HTTP request by using `wp_add_inline_script()` eg:

```php
wp_add_inline_script(
  'altis-consent',
  sprintf( 'var %1$s = window.%1$s || [];' . 
    '%1$s.push( { event: 'altis-consent-changed', altisConsent: Altis.Consent.getCategories().join( \', \' ) } );' .
    'document.addEventListener( \'wp_listen_for_consent_change\', function () { %1$s.push( { event: 'altis-consent-changed', altisConsent: Altis.Consent.getCategories().join( \', \' ) } ) } );',
    apply_filters( 'hm_gtm_data_layer_var', 'dataLayer' )
  ),
  'after'
);
```

**Review by roborourke — APPROVED** [review]

File: docs/consent/google-tag-manager.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Google Tag Manager Consent Integration
+
+Support for Google Tag Manager is provided via GTM's data layer variable. When either the page loads or consent is subsequently changed the data layer is updated allowing you to create triggers based on the consent given.
```
Comment (jazzsequence) [inline docs/consent/google-tag-manager.md:3]: If we're talking about a variable, maybe we should name it specifically? e.g. "via GTM's `dataLayer` variable."

File: docs/consent/google-tag-manager.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+The event name is `altis-consent-changed` and the variable is named `altisConsent`. The variable data is a list of the [consent categories](./Consent-API.md#consent-categories) currently given for example "functional, statistics-anonymous".
```
Comment (jazzsequence) [inline docs/consent/google-tag-manager.md:5]: I'd put "for example" in parentheses

**Comment by jazzsequence** [comment]
Have a couple thoughts on the docs. I will go through this process again for QA and make sure it all makes sense to me and can update the docs then.

**Pull request merged**